src/gennylib/include/gennylib/conventions.hpp

```diff
@@ -17,26 +21,253 @@ void decodeNodeInto(T& out, const YAML::Node& node, const S& fallback) {
     out = node.as<T>(fallback);
 }
 
+/**
+ * Intermediate state for converting YAML syntax into a native integer type of your choice.
+ *
+ * size_t is used by default; smaller types can be explicitly converted to as needed.
+ */
+struct UIntSpec {
+    UIntSpec() = default;
+    ~UIntSpec() = default;
+
+    explicit UIntSpec(size_t v) : value{v} {}
+    // size_t is used by default, you can explicitly cast to another type if needed.
+    size_t value;
```

Comment:
There may be a missed opportunity here to bake optionality into the config types. There is no default ctor and no bool operator, so this could have tech value.